### ISSUE TITLE
AI Image Classifier for Google Colab

### ISSUE THREAD

**Issue by kshareefbasha** [kshareefbasha/image-processing#1]
# Image Classification in Google Colab

This project allows users to upload an image to Google Colab and perform image classification using a pre-trained ResNet50 model. The top 3 predicted classes for the uploaded image are displayed.

### Prerequisites

- TensorFlow 2.x
- Keras
- numpy

### Environment Setup

1. Install the required libraries in your Google Colab environment:

```python
!pip install tensorflow
```

2. Clone this repository to your Colab environment:

```python
!git clone https://github.com/kshareefbasha/image-processing.git
```

## Usage

1. Upload an image to Google Colab by running the following code:

```python
from google.colab import files
uploaded = files.upload()
```

2. Get the file name of the uploaded image:

```python
image_filename = list(uploaded.keys())[0]
```

3. Run the image classification code:

```python
!python image_classification.py
```

This will provide the top 3 predicted classes for the uploaded image.

## Example Output

```
1: golden_retriever (0.87)
2: Labrador_retriever (0.05)
3: kuvasz (0.03)
```

## License

This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details.

## Improved image_classification.py

The project includes an improved version of `image_classification.py` with a more structured and user-friendly code:

```python
# Improved image_classification.py

import tensorflow as tf
from tensorflow.keras.preprocessing import image
from tensorflow.keras.applications.resnet50 import preprocess_input, decode_predictions
import numpy as np

# Load the pre-trained ResNet50 model
model = tf.keras.applications.ResNet50(weights='imagenet')

# Define a function for image classification
def classify_image(image_path):
    # Load and preprocess the image
    img = image.load_img(image_path, target_size=(224, 224))
    x = image.img_to_array(img)
    x = np.expand_dims(x, axis=0)
    x = preprocess_input(x)

    # Make predictions
    predictions = model.predict(x)
    decoded_predictions = decode_predictions(predictions, top=3)[0]

    # Print the top 3 predicted classes
    for i, (imagenet_id, label, score) in enumerate(decoded_predictions):
        print(f"{i + 1}: {label} ({score:.2f})")

# Get the file name of the uploaded image
image_filename = input("Enter the path to the image file: ")

# Perform image classification with the uploaded image
classify_image(image_filename)
